test/agenda.js

```diff
@@ -568,4 +568,41 @@ describe('Agenda', () => {
       expect(job3.attrs.data).to.be(1);
     });
   });
+
+  describe('process jobs', function() { // eslint-disable-line prefer-arrow-callback
+    it('should not cause unhandledRejection', async function() {
+      // This unit tests if for this bug [https://github.com/agenda/agenda/issues/884]
+      // which is not reproducible with default agenda config on shorter processEvery.
+      // Thus we set the test timeout to 10000, and the delay below to 6000.
+      this.timeout(10000);
+
+      const unhandledRejections = [];
+      const rejectionsHandler = error => unhandledRejections.push(error);
+      process.on('unhandledRejection', rejectionsHandler);
+
+      let j1processes = 0;
+      jobs.define('j1', (job, done) => {
+        j1processes += 1;
+        done();
+      });
+
+      let j2processes = 0;
+      jobs.define('j2', (job, done) => {
+        j2processes += 1;
+        done();
+      });
+
+      await jobs.start();
+      await jobs.every('5 seconds', 'j1');
+      await jobs.every('10 seconds', 'j2');
+
+      await delay(6000);
```

Comment:
Why this long delay? Could the test be structured without it (or with shorter?)

Edit: oh I saw your code-comment above. Still would love if we'd have something else than timeout to test.

Reply:
I tried shortening all the delays x10. For some reason the bug didn't reproduce. Not sure why.
It's likely can be shortened. I just didn't have much time to investigate. Sorry.